docs/docs/understanding/agent/structured_output.md

```diff
@@ -0,0 +1,185 @@
+# Using Structured Output
+
+Agents are knowingly unreliable in their outputs, as most of the times we would like them to answer in a specific format but they do not follow the rules due to their internal complexity.
+
+To tackle this challenge, we added the possibility of specifying a structured output format within single and multi-agent workflows so that you can have a more fine-grained control over the data structure that your agent returns.
+
+You can do it by adding an `output_cls` argument to the initialization options of your agent/multi-agent workflows:
```

Comment:
Some of the phrasing in this into seems a little passive and confusing. Could it be made more direct and expositional?

```
Most of the time you need results from an agent in a specific format. Agents results can return structured json in 2 ways:
1. `output_cls` – a pydantic model to use as a schema for the output
2. `structured_output_fn` – For more advanced use cases, supply a custom function that validates or rewrites the agent’s conversation into any model you want.

Both single-agent `FunctionAgent` and multi-agent `AgentWorkflow` workflows support these options.
```